src/sentry/models/apiauthorization.py

```diff
@@ -36,6 +37,18 @@ class Meta:
         app_label = "sentry"
         db_table = "sentry_apiauthorization"
         unique_together = (("user", "application"),)
+        constraints = [
+            models.UniqueConstraint(
+                fields=["user", "application"],
+                name="apiauthorization_user_app",
+                condition=Q(organization_id__isnull=True),
+            ),
+            models.UniqueConstraint(
+                fields=["user", "application", "organization_id"],
+                name="apiauthorization_user_app_org",
+                condition=~Q(organization_id__isnull=True),
```

Comment:
Nit: The double negative here is a little confusing, I'd just use `Q(organization_id__isnull=False)`